src/main/java/com/firstpenguin/tinyurl/restservice/controller/TinyURLController.java

```diff
@@ -34,19 +36,27 @@ public Url shortUrl(@RequestBody Url url) throws Exception {
         // do we have the URL already stored?
         // TODO: move this call to the Cache
         Url urlFromRepo = urlRepository.findByLongUrlHash(sha1);
+        
+        if (urlFromRepo != null) {
```

Comment:
Good to pull the latest. Code is simplified based on your last review.

Reply:
Pulled and merged.